omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -1049,27 +1046,27 @@ def _move_hand_direct_joint(self, joint_pos, stop_on_contact=False, ignore_failu
         Yields action for the robot to move its arm to reach the specified joint positions by directly actuating with no planner
 
         Args:
-            joint_pos (np.array): Array of joint positions for the arm
+            joint_pos (th.tensor): Array of joint positions for the arm
             stop_on_contact (boolean): Determines whether to stop move once an object is hit
             ignore_failure (boolean): Determines whether to throw error for not reaching final joint positions
 
         Returns:
-            np.array or None: Action array for one step for the robot to move arm or None if its at the joint positions
+            th.tensor or None: Action array for one step for the robot to move arm or None if its at the joint positions
         """
         controller_name = f"arm_{self.arm}"
         use_delta = self.robot._controllers[controller_name].use_delta_commands
 
         # Store the previous eef pose for checking if we got stuck
-        prev_eef_pos = np.zeros(3)
+        prev_eef_pos = th.zeros(3)
 
         for _ in range(m.MAX_STEPS_FOR_HAND_MOVE_JOINT):
             current_joint_pos = self.robot.get_joint_positions()[self._manipulation_control_idx]
-            diff_joint_pos = np.array(joint_pos) - np.array(current_joint_pos)
-            if np.max(np.abs(diff_joint_pos)) < m.JOINT_POS_DIFF_THRESHOLD:
+            diff_joint_pos = th.tensor(joint_pos) - th.tensor(current_joint_pos)
+            if th.max(th.abs(diff_joint_pos)).values < m.JOINT_POS_DIFF_THRESHOLD:
```

Comment:
for `th.max`, if you don't pass in the `dim` the argument, the returned value is just a torch tensor, so you cannot call `.values` on it. This code is actually wrong. You should refactor all mentions of `th.max`.